components/datetime/src/pattern/mod.rs

```diff
@@ -22,7 +22,8 @@ pub use item::{GenericPatternItem, PatternItem};
 )]
 #[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-pub(crate) enum TimeGranularity {
+#[non_exhaustive]
```

Comment:
Nit: If `TimeGranularity` is in the data struct, I guess it should be exhaustive?

Recall our discussion from last week where we said that if we needed to add more fields to this in the future, on a major version bump, we can safely re-create the old enum under a different name and use the current name for the new enum.